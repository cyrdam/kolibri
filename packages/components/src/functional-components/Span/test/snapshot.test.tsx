import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';
import KolSpanFc from './..';

describe('KolSpanFc', () => {
	it('should render with default props', async () => {
		const page = await newSpecPage({
			components: [KolSpanFc],
			template: () => <KolSpanFc label="Default Label" />,
		});

		expect(page.root).toBeDefined();
		expect(page.root).toMatchSnapshot();
		expect(page.root?.tagName.toLowerCase()).toBe('span');
		expect(page.root?.classList.contains('kol-span')).toBe(true);
		expect(page.root?.textContent).toContain('Default Label');
	});

	it('should render with icons', async () => {
		const icons = { top: { icon: 'top-icon' }, left: { icon: 'left-icon' }, right: { icon: 'right-icon' }, bottom: { icon: 'bottom-icon' } };
		const page = await newSpecPage({
			components: [KolSpanFc],
			template: () => <KolSpanFc label="Label with Icons" icons={icons} />,
		});

		expect(page.root).toBeDefined();
		expect(page.root).toMatchSnapshot();
		expect(page.root?.querySelector('.top')).not.toBeNull();
		expect(page.root?.querySelector('.left')).not.toBeNull();
		expect(page.root?.querySelector('.right')).not.toBeNull();
		expect(page.root?.querySelector('.bottom')).not.toBeNull();
	});

	it('should hide label when hideLabel is true', async () => {
		const page = await newSpecPage({
			components: [KolSpanFc],
			template: () => <KolSpanFc label="Hidden Label" hideLabel={true} />,
		});

		expect(page.root).toBeDefined();
		expect(page.root).toMatchSnapshot();
		expect(page.root?.classList.contains('kol-span--hide-label')).toBe(true);
		expect(page.root?.textContent).not.toContain('Hidden Label');
	});

	it('should render with badge text', async () => {
		const page = await newSpecPage({
			components: [KolSpanFc],
			template: () => <KolSpanFc label="Label with Badge" badgeText="Badge" />,
		});

		expect(page.root).toBeDefined();
		expect(page.root).toMatchSnapshot();
		expect(page.root?.textContent).toContain('Badge');
	});

	it('should allow markdown in label', async () => {
		const page = await newSpecPage({
			components: [KolSpanFc],
			template: () => <KolSpanFc label="**Bold Label**" allowMarkdown={true} />,
		});

		expect(page.root).toBeDefined();
		expect(page.root).toMatchSnapshot();
		// Here you could add specific markdown rendering checks if needed
	});
});
