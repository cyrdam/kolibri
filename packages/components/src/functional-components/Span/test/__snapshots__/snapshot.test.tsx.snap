// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KolSpanFc should allow markdown in label 1`] = `
<span class="kol-span">
  <span class="kol-span__container">
    <span class="kol-span__label md">
      <strong>
        Bold Label
      </strong>
    </span>
    <span aria-hidden="true" class="kol-span__label" hidden=""></span>
  </span>
</span>
`;

exports[`KolSpanFc should hide label when hideLabel is true 1`] = `
<span class="kol-span kol-span--hide-label">
  <span class="kol-span__container">
    <span aria-hidden="true" class="kol-span__label" hidden=""></span>
  </span>
</span>
`;

exports[`KolSpanFc should render with badge text 1`] = `
<span class="kol-span">
  <span class="kol-span__container">
    <span class="kol-span__label">
      Label with
      <u>
        Badge
      </u>
    </span>
    <span aria-hidden="true" class="kol-span__label" hidden=""></span>
    <span aria-hidden="true" class="badge-text-hint">
      Badge
    </span>
  </span>
</span>
`;

exports[`KolSpanFc should render with default props 1`] = `
<span class="kol-span">
  <span class="kol-span__container">
    <span class="kol-span__label">
      Default Label
    </span>
    <span aria-hidden="true" class="kol-span__label" hidden=""></span>
  </span>
</span>
`;

exports[`KolSpanFc should render with icons 1`] = `
<span class="kol-span">
  <kol-icon _icons="top-icon" _label="" class="icon top"></kol-icon>
  <span class="kol-span__container">
    <kol-icon _icons="left-icon" _label="" class="icon left"></kol-icon>
    <span class="kol-span__label">
      Label with Icons
    </span>
    <span aria-hidden="true" class="kol-span__label" hidden=""></span>
    <kol-icon _icons="right-icon" _label="" class="icon right"></kol-icon>
  </span>
  <kol-icon _icons="bottom-icon" _label="" class="bottom icon"></kol-icon>
</span>
`;
